lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java

```diff
@@ -156,35 +146,72 @@ public void setMethodInstance(Method m) {
         return answer;
     }
 
-    private ActionParameters detectFieldActionParameters(Class<?> actionClazz) throws ClassNotFoundException {
+    private ActionParameters detectFieldActionParameters(Class<?> actionClazz) {
         ActionParameters params = null;
-        for (Method method : actionClazz.getMethods()) {
-            // Find setters to avoid the get / is confusion
-            if (method.getParameterCount() == 1 && method.getName().startsWith("set")) {
-                // We have a parameter
-                if (params == null) {
-                    params = new ActionParameters();
+        for (java.lang.reflect.Field field : actionClazz.getDeclaredFields()) {
+            AtlasFieldActionParameter anno = field.getAnnotation(AtlasFieldActionParameter.class);
+            if (anno == null) {
+                continue;
+            }
+
+            if (field.getType() != String.class) {
+                LOG.debug("Field action parameters must be declared of type String;"
```

Comment:
We should support other atlas primitive types but not only String. Even when `type` is not defined in annotation, we should autodetect from `java.lang.Field.getType()`.

Reply:
This is in preparation for variable support. Since the intention is for all parameters to support variables, they will all need to define any parameters as type String in the class to allow for references to the variables (e.g., "$<variableName>$"), while setting the 'type' in the annotation to the actual type to be stored in the output (In the case of my StoreOutput transformation, it just happens to be the default String type anyways). Since this is a new annotation and therefore is currently unused outside of my example, it won't affect any of the current code or custom actions, so I included the check here.

Reply:
Do not request to merge your personal preparation to the upstream master. This is really disrespectful to the project.

It's ugly if user have to declare all the custom action parameters as a String just to accept variables. Variable resolution just have to happen before it's set onto action class.
Also it makes more sense to replace current xsd based action parameters with this annotation completely, so OOTB ones also have to be addressed. It doesn't make sense to be specialized to custom action.